File: internal/dto/importer.go
package dto

import "github.com/davidmovas/postulator/internal/infra/importer"

type ImportResult struct {
	TotalRead    int      `json:"totalRead"`
	TotalAdded   int      `json:"totalAdded"`
	TotalSkipped int      `json:"totalSkipped"`
	Added        []string `json:"added,omitempty"`
	Skipped      []string `json:"skipped,omitempty"`
	Errors       []string `json:"errors,omitempty"`
}

func NewImportResult(entity *importer.ImportResult) *ImportResult {
	d := &ImportResult{}
	return d.FromEntity(entity)
}

func (d *ImportResult) FromEntity(entity *importer.ImportResult) *ImportResult {
	if entity == nil {
		return d
	}
	d.TotalRead = entity.TotalRead
	d.TotalAdded = entity.TotalAdded
	d.TotalSkipped = entity.TotalSkipped
	d.Added = entity.Added
	d.Skipped = entity.Skipped
	d.Errors = entity.Errors
	return d
}

type ImportTopicsRequest struct {
	FilePath string `json:"filePath"`
}

type ImportAndAssignRequest struct {
	FilePath string `json:"filePath"`
	SiteID   int64  `json:"siteId"`
}

File: frontend/src/models/importer.ts
import { dto } from "@/wailsjs/wailsjs/go/models";

export interface ImportResult {
    totalRead: number
    totalAdded: number
    totalSkipped: number
    added: string[]
    skipped: string[]
    errors: string[]
}

export function mapImportResult(x: dto.ImportResult): ImportResult {
    return {
        totalRead: x.totalRead,
        totalAdded: x.totalAdded,
        totalSkipped: x.totalSkipped,
        added: x.added || [],
        skipped: x.skipped || [],
        errors: x.errors || [],
    };
}


